src/main/java/org/jenkinsci/plugins/workflow/graph/BlockStartNode.java

```diff
@@ -51,4 +56,23 @@ protected BlockStartNode(FlowExecution exec, String id, List<FlowNode> parents)
     public BlockEndNode getEndNode() {
         return this.getExecution().getEndNode(this);
     }
+
+    /** Get the immedate children of this block. */
+    @Nonnull
+    public List<FlowNode> getImmediateChildren() {
+        return this.getExecution().getImmediateChildrenForBlockStart(this);
+    }
+
+    @CheckForNull
+    @Override
+    public Result getStatus() {
+        BlockEndNode endNode = getEndNode();
+        // Only check for errors or explicit status if this block is complete.
+        if (endNode != null) {
+            return endNode.getStatus();
```

Comment:
Will this handle a block that may itself may fail when run?  I.E. Dir step that throws IOException?

Reply:
Good question! Needs a test.

Reply:
And now has one.